runtime/lua/vim/treesitter/dev.lua

```diff
@@ -159,7 +159,10 @@ end
 local function set_dev_properties(w, b)
   vim.wo[w].scrolloff = 5
   vim.wo[w].wrap = false
-  vim.wo[w].foldmethod = 'manual' -- disable folding
```

Comment:
I'm not sure this worked because I get folding.

Reply:
I don't really know but that's possibly because `foldmethod` set during startup is before this line

Reply:
So does that mean the options below are also ignored?

Reply:
I think it's because `filetype` is set last. We may want to set that first?

Reply:
I don't think there was much intent, it just didn't matter before (@gpanders ?); setting filetype first and _then_  adding options certainly makes sense.

Reply:
No. This is still set. The code also doesn't fold until one press zc/zC/za/zA

Reply:
I think it's because I set treesitter folds in a filetype autocmd.

Reply:
Running `nvim --clean src/nlua0.c -c` doesn't fold automatically. So I don't think this is an issue with the PR

Reply:
Yes, but it might be a problem with typical setups.

Applying options in filetype autocmds is a pretty common (and recommended?) idiom, so whatever we do here needs to take that into account. Whether we override user config or not should be an intended behaviour.

Reply:
>Applying options in filetype autocmds is a pretty common (and recommended?) idiom, so whatever we do here needs to take that into account. Whether we override user config or not should be an intentional behaviour.

My gut says we should set the "default" settings here which can be overriden by a filetype autocommand or ftplugin, if desired. Which means this PR is correct as-is.

Reply:
It would be nice if the code showed this intent better.

A simple comment would do.